pytools/__init__.py

```diff
@@ -196,6 +196,12 @@
 
 .. autofunction:: unique
 
+SQLite-related functions
+------------------------
+
+.. autofunction:: get_sqlite3_thread_safety_level
+.. autofunction:: is_sqlite3_fully_threadsafe
```

Comment:
I'm not sure I'd like to document these, since it encodes things that feel like implementation details. Maybe push this off into `pytools._sqlite`?

Reply:
Moving these functions into a separate module makes sense I think, but I thought these functions are pretty generic? Which implementation details do they encode?

Reply:
> How cpython and threads and sqlite interact. The details of how this works at the moment is kind of insane ATM

I see - is this in reference to the translation between thread support levels in https://github.com/inducer/pytools/pull/234/commits/07412b2b08f63e7c6da8af4bc2424aa5856d0d10#diff-2d67e0c2f525bd7f05e4909f1248231d4b1f3b92d2f1be94487d5b2d4cb46b78R2992-R2996, or are there other problems in interacting between threads/sqlite? If so, could those be the issue for the failures I mentioned in https://github.com/inducer/pytools/pull/234#issuecomment-2159452812?